github/resource_github_repository_ruleset.go

```diff
@@ -0,0 +1,550 @@
+)
+
+func resourceGithubRepositoryRuleset() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGithubRepositoryRulesetCreate,
+		Read:   resourceGithubRepositoryRulesetRead,
+		Update: resourceGithubRepositoryRulesetUpdate,
+		Delete: resourceGithubRepositoryRulesetDelete,
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+
+		SchemaVersion: 1,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 100),
+				Description:  "The name of the ruleset.",
+			},
+			"target": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Possible values are `branch` and `tag`.",
+			},
+			"repository": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Name of the repository to apply rulset to.",
+			},
+			"enforcement": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"disabled", "active", "evaluate"}, false),
+				Description:  "Possible values for Enforcement are `disabled`, `active`, `evaluate`. Note: `evaluate` is currently only supported for owners of type `organization`.",
+			},
+
+			"bypass_actors": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "The actors that can bypass the rules in this ruleset.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"actor_id": {
+							Type:        schema.TypeInt,
+							Required:    true,
+							Description: "The ID of the actor that can bypass a ruleset",
+						},
+						"actor_type": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "The type of actor that can bypass a ruleset. Can be one of: `RepositoryRole`, `Team`, `Integration`, `OrganizationAdmin`.",
+						},
+						"bypass_mode": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.StringInSlice([]string{"always", "pull_request"}, false),
+							Description:  "When the specified actor can bypass the ruleset. pull_request means that an actor can only bypass rules on pull requests. Can be one of: `always`, `pull_request`.",
+						},
+					}},
+			},
+			"node_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "GraphQL global node id for use with v4 API.",
+			},
+			"ruleset_id": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "GitHub ID for the ruleset.",
+			},
+			"conditions": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				MaxItems:    1,
+				Description: "Parameters for a repository ruleset ref name condition.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"ref_name": {
```

Comment:
I think this one might be useful to keep in case it’s an org ruleset, those have `include` and `exclude` if you use `repository_name`, so it could be more clear to have `ref_name` there so the inner parameters aren’t at the top level and also in `repository_name`, but I have no strong preference here

Reply:
Fair point. Now that I've looked at the org level one a bit, I think it's fine to keep the `ref_name` there

Reply:
Sounds good, we can keep this in mind and then check with others once we’re ready to have this reviewed

Reply:
Sounds good to me